DigitalPuzzle/DigitalPuzzle.ino

```diff
@@ -1,349 +1,238 @@
-          break;
-        default:  //
-          ReaderResult[0] = 0; 
-          break;
-      }
-    case 1:
-      switch(CompareNFC(ReadRFID(1)))
-      {
-        case 1:   //AND GATE
-          ReaderResult[1] = Check_AND(StartValues[2], StartValues[3]);
-          break;
-        case 2:   //OR GATE
-          ReaderResult[1] = Check_OR(StartValues[2], StartValues[3]);
-          break;
-        case 3:   //NAND GATE
-          ReaderResult[1] = Check_NAND(StartValues[2], StartValues[3]);
-          break;
-        case 4:   //NOR GATE
-          ReaderResult[1] = Check_NOR(StartValues[2], StartValues[3]);
-          break;
-        case 5:   //XOR GATE
-          ReaderResult[1] = Check_XOR(StartValues[2], StartValues[3]);
-          break;
-        default:  //
-          ReaderResult[1] = 0; 
-          break;
-      }
-  }
-
-  return ReaderResult;
-}
-
-/*************************************************************
- * Function：ApplyResults
- * Description：turns the ledstrips on and off, depending on
-   the result. 
- * Input parameter：none
- * Return：returns none
- *************************************************************/
-void ApplyResults()
-{
-  if (GetReaderResult(0)[0] == true)
-  {
-    digitalWrite(GreenPins[0], HIGH);
-    digitalWrite(RedPins[0], LOW);
-  }
-  else
-  {
-    digitalWrite(GreenPins[0], LOW);
-    digitalWrite(RedPins[0], HIGH);
-  }
-
-  if (GetReaderResult(1)[1] == true)
-  {
-      digitalWrite(GreenPins[1], HIGH);
-      digitalWrite(RedPins[1], LOW);
-  }
-  else
-  {   digitalWrite(GreenPins[1], LOW);
-      digitalWrite(RedPins[1], HIGH);
-  }
+	switch (ReaderSelect) //Select one of the eight readers 
+	{                     //Pin Configuration of RFID.begin (IRQ_PIN,SCK_PIN,MOSI_PIN,MISO_PIN,SDA_PIN,RST_PIN)
+	case 0:
+		rfid.begin(12, 13, 9, 24, 10, 11);
+		break;
+	case 1:
+		rfid.begin(12, 13, 9, 26, 10, 11);
+		break;
+	case 2:
+		rfid.begin(12, 13, 9, 28, 10, 11);
+		break;
+	case 3:
+		rfid.begin(12, 13, 9, 30, 10, 11);
+		break;
+	case 4:
+		rfid.begin(12, 13, 9, 22, 10, 11);
+		break;
+	case 5:
+		rfid.begin(12, 13, 9, 34, 10, 11);
+		break;
+	case 6:
+		rfid.begin(12, 13, 9, 32, 10, 11);
+		break;
+	case 7:
+		rfid.begin(12, 13, 9, 36, 10, 11);
+		break;
+	}
+	delay(10);  //delay 100ms
+	rfid.init(); //initialize the RFID
+
+	uchar status;
+	uchar str[MAX_LEN];
+	// Search card, return card types
+	status = rfid.request(PICC_REQIDL, str);
+	if (status != MI_OK)
+	{
+		for (int i; i < 5; i++)
+		{
+			IDStorage[i] = 0; //Set IDStorage to 0 when no card is detected
+		}
+		return IDStorage;
+	}
+	//Prevent conflict, return the 4 bytes Serial number of the card
+	status = rfid.anticoll(str);
+	if (status == MI_OK)
+	{
+		memcpy(IDStorage, str, 5);
+		Serial.print("Reader ");
+		Serial.print(ReaderSelect);
+		Serial.print(": ");
+		rfid.showCardID(IDStorage);//show the card 
+		Serial.println();
+		delay(10);  //delay of 100ms
```

Comment:
10 of 100?

Reply:
Whoops I did it again! 
- Bram 20-06-2019 (& Brittney Spears 16-05-2000)